File: ds/stack/mystack.cpp
#include <iostream>
#include <vector>
using namespace std;

class myStack {
private:
    vector<string> items;

public:
    void push(string item) {
        items.push_back(item);
    }

    string pop() {
        if (is_empty()) {
            return "";
        }
        string item = items.back();
        items.pop_back();
        return item;
    }

    bool is_empty() {
        return items.size() == 0;
    }
};

int main() {
    myStack s;
    s.push("a");
    s.push("b");
    s.push("c");

    cout << s.pop() << endl;  // c
    cout << s.pop() << endl;  // b
    cout << s.pop() << endl;  // a

    return 0;
}

File: ds/stack/mystack.js
class myStack {
    constructor() {
        this.items = [];
    }

    push(item) {
        this.items.push(item);
    }

    pop() {
        if (this.is_empty()) {
            return null;
        }
        return this.items.pop();
    }

    is_empty() {
        return this.items.length === 0;
    }
}

const s = new myStack();
s.push('a');
s.push('b');
s.push('c');

console.log(s.pop());  // c
console.log(s.pop());  // b
console.log(s.pop());  // a
